Build system fixes: use include/library paths from pkg-config, and strip leading 'v' from librtlsdr version

agrif opened the pull request:
This PR fixes two issues with the CMake build scripts I encountered on my Mac:

 1. CMake now uses the include and library paths reported by pkg-config, which lets it build successfully even when (e.g.) librtlsdr is installed in a nonstandard place.
 2. The bias tee version check for librtlsdr now looks for a leading 'v' on the version, and strips it if necessary. Some versions of librtlsdr include this 'v' (as when building from source), and some do not (as in the Debian package). If the version includes a 'v' without being stripped first, the bias tee check fails and I cannot use a bias tee even though my version of librtlsdr supports it. This change supports both version formats.

I'm happy to make any changes necessary to help get this merged!

I also have a few EMWIN handler fixes. I'll clean those up and probably open a separate PR once they're ready.

pietern reviewed (commented):
Thanks for submitting the PR!

Are you able to build on macOS with these changes? If so, what do you use for the dependencies (e.g. homebrew, ports, diy)?

agrif commented:
Yes! I'm building this successfully on MacOS using librtlsdr and opencv from macports. I've only used `goesrecv` but it seems to work fine.

It looks like `target_link_directories` is too new for the targeted cmake version. There's a couple of solutions:

 * Use absolute library paths from pkg-config rather than a fixed list of library names (if it provides them, I would have to check)
 * Use a version check to [switch between `target_link_directories` and `link_directories`, like here.](https://github.com/ros2/libyaml_vendor/pull/29/commits/d7aee26adafa71529d344a3be84106a753780443)
 * Just use `link_directories`.

I'll try absolute paths first, but I'll have to test it out on my pi and elsewhere to make sure it doesn't break any builds.

agrif commented:
I've updated the pkg-config sections to use absolute library paths from `${FOO_LINK_LIBRARIES}`, [as recommended by the CMake documentation](https://cmake.org/cmake/help/latest/command/target_link_directories.html). This should make the changes compatible with cmake < 3.13. I've tested these changes on my mac, x64 debian, and raspbian. Unfortunately this means linking with *all* of OpenCV instead of selectively, but I don't think that hurts much.

The pkgconfig file for airspy puts the `libairspy` directory in the include path, so I've changed the source to include `airspy.h` directly. This was also tested and works on all three systems I have access to.

I've also noted which versions of librtlsdr have the pesky leading 'v'.
